### PR TITLE
DOC: Fix empty 'C style guide' page

### PR DESCRIPTION
**Pull request by bjnath**
The page had been left empty when C_STYLE_GUIDE.rst.txt was migrated to NEP 45. 

Added link to NEP 45. Fixed uppercasing in title.

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
Alternatively we could consider replacing the TOC entry with an external link, i.e. in `dev/index.rst` replace `style_guide` with `NumPy C style guide <https://numpy.org/neps/nep-0045-c_style_guide.html>` (AFAIK sphinx doesn't support roles in the toctree entries, only external links)

The advantage being that the redirection will be automatic and there's no need to click through (we can also then delete style_guide.rst).

Note that this approach would break any internal `style_guide` links, which would then need to be redirected (via intersphinx) to NEP45.

**Comment by bjnath** [comment]
You explain the tradeoffs well. 

This may be an example of a fix that's needed urgently now and can be done up nicely later.

**Comment by bjnath** [comment]
It's still a worthwhile change. If this PR is merged, I'll open an issue saying the linking should be improved and quoting your directions.

**Comment by rossbar** [comment]
Rather than create multiple issues/PRs, might as well do it all at once - the change turns out to be very minor as a `grep` reveals that the link to `style_guide` was not used anywhere else in the docs.

**Review by eric-wieser — APPROVED** [review]
Perhaps worth filing a sphinx issue about intersphinx links in TOCs?

File: doc/source/dev/index.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ Contributing to NumPy
    development_environment
    development_workflow
    ../benchmarking
-   style_guide
+   NumPy C style guide <https://numpy.org/neps/nep-0045-c_style_guide.html>
```
Comment (eric-wieser) [inline doc/source/dev/index.rst:16]: I wonder... Does `nep:NEP45` work?

Comment (rossbar) [reply doc/source/dev/index.rst:16]: I just gave it a try and this (and several other iterations) didn't work. AFAICT `toctree` only supports using relative paths or external links, but does not parse roles (or role-like targets) for the entries.

**Comment by rossbar** [comment]
> Perhaps worth filing a sphinx issue about intersphinx links in TOCs?

Looks like it's been raised: https://github.com/sphinx-doc/sphinx/issues/1836

File: doc/source/dev/index.rst (hunk 1 above)
Comment (rossbar) [reply doc/source/dev/index.rst:16]: Oops, that's what I meant by "role-like target" (I'm not fluent with sphinx terminology). Using `NumPy C style guide <neps:NEP45>` or just `neps:NEP45` results in:
```
WARNING: toctree contains reference to nonexisting document 'dev/neps:NEP45'
```

**Comment by rossbar** [comment]
Thanks @bjnath !

**Comment by bjnath** [comment]
Thanks back at you, @rossbar !  Much better your way.

**Pull request merged**